src/persist-client/src/impl/metrics.rs

```diff
@@ -37,6 +37,8 @@ pub struct Metrics {
     pub cmds: CmdsMetrics,
     /// Metrics for each retry loop.
     pub retries: RetriesMetrics,
+    /// Metrics for various encodings and decodings.
+    pub codec: CodecMetrics,
```

Comment:
oh! I should make this a set of vec counters with state, key, val as the labels. also I should add trace batch encode/decode to this PR